README_turkish.md

```diff
@@ -0,0 +1,152 @@
+# JTAG to AXI Master IP
+
+Xilinx'in `JTAG to AXI Master` isimli bir IP'si var. Literatürde gördüğüm kadarıyla yaygınca kullanılan bir IP değil. Esasında debug amaçlı tasarlanmış bir IP. Özellikle birden fazla adres alanına `AXI-Lite` veya `AXI Memory Mapped` arayüzü ile tek bir yerden erişmek için kullanılıyor. Aşağıda JTAG to AXI Master IP'sinin proje içerisinde nasıl duracağı ile ilgili örnek bir figür var:
+
+![capture0](assets/figure1.drawio.png)
+
+JTAG to AXI Master IP'sinin **adres alanı (address space)** bulunmuyor. Bu yüzden istediği adrese yazma, istediği adresten okuma işlemini yapabiliyor. Bunu yaparken başka bir IP'yi projeden çıkarmamıza gerek kalmıyor.
+
+Temelde, bu IP sayesinde Vivado harici herhangi bir programa gerek kalmadan; FPGA programlandıktan sonra `TCL konsolu` ile yukarıdaki görseldeki yapıyı koruyarak istediğimiz AXI-Lite veya AXI Memory Mapped slave arayüzüne erişebilmekteyiz.
+
+## Motivasyon
+
+Mesela; proje gereği **AXI IIC**, **AXI Quad SPI** ve **AXI Bram Controller** IP'lerini kullanıyoruz. Bu durumda her biri için bir adres **offset'i** belirleniyor (Vivado veya kullanıcı tarafından). Bu adresler dikkate alınarak projenin veya IP'lerin gereği olarak ilgili kısımlara veriler yazılır. Fakat henüz geliştirme aşamasında küçük denemeler yapabilmek için ilgili IP'ye AXI-Lite veya AXI Memory Mapped üzerinden erişmemiz ve yazma/okuma yapmamız gerekiyor. İşte JTAG to AXI Master IP'si sayesinde projeye dahil IP'lerin ilgili **AXI arayüzlerine** erişim sağlayabiliyoruz. Bunu yapmak için **Vitis** veya eski ismiyle **Xilinx SDK** veya işletim sistemi gibi kompleks ve başka programlara ihtiyaç duyan çözümlere gerek olmuyor.
+
+## Diğer Alternatifler
+
+Diğer alternatif çözümlerden de bahsetmek gerekir. Aşağıda diğer çözümler konusunda kısa anlatımlar yer alıyor:
+
+- Xilinx Vitis (eski adıyla Xilinx SDK) programının **xsct** desteği bulunmakta. Bunun için `MicroBlaze Debug Module (MDM)` IP'sine ihtiyaç var. Çeşitli işlemler sonrası `mwr` veya `mrd` komutlarıyla ilgili adreslere gerekli veriler yazılabilir.
```

Comment:
MDM'yi microblaze olmadan hiç kullanmadım. Yani tek başına sırf AXI iletişimi için kullanılabilir mi bilmiyorum. Eğer olmuyorsa ublaze gerektirmesi bir dezavantaj olarak da görülebilir bence. Ama tahminim tek başına çalışacağı yönündedir, denenebilir.

Reply:
Evet, bunu ben de denemek istiyorum. Belki bu yazıyı yayınlamadan bir bakabilirim. Hatta onu ayrıca kısa bir yazı olarak da yazabilirim eğer oluyorsa.